packages/ddp-client/client/queueStubsHelpers.js

```diff
@@ -107,7 +107,7 @@ export const loadAsyncStubHelpers = () => {
         });
 
         Meteor._setImmediate(() => {
-          if (!finished) {
+          if (!finished && Meteor.isDebug) {
```

Comment:
This warning is important for developers. If it is logged, it means that it is no longer safe to call methods whenever you want to, and you have to either fix the method stub, or start being careful with using certain Meteor api's like you had to be with Meteor 2 since app/package code could run during an async method stub.

It should be very rarely shown. If it is appearing enough that you want to disable it by default, there might be a bug with when finished it set - it should be set right after (within the same macrotask) of when the method stub finishes.

Reply:
Do you mean there could be a bug in this queue where we go to the next before the previous is done? Or is it the problem in the developer's app?

I am not 100%, but I think one way to reproduce this is just by doing the following:

```
Meteor.callAsync("something")
Meteor.callAsync("somethingElse")
```

But this queue should make this be possible, right?

How did make the log show up on your example, @nachocodoner?

Reply:
If it happens with every method call, then it is a bug with when `finished` is set to `true`. I'm guessing it is set after it receives the server result instead of after the stub finishes. This also would be a bug with the queue - its purpose is to run one method stub at a time, so there is no need to wait to receive the server result. Waiting defeats the purpose of calling `this.unblock` on the server.

We probably should add a test for this. Maybe something like:
```
// server
let serverEvents = [];
Meteor.methods({
  async unblockedMethod() {
     serverEvents.push('unblock start');
     this.unblock();
     await sleep(200);
     serverEvents.push('unblock end')
  },
  blockingMethod() {
    serverEvents.push('blockingMethod');
  }
});

// client (server methods should also have a no-op client stub)

Meteor.callAsync('unblockedMethod');
Meteor.callAsync('blockingMethod');
// check serverEvents to ensure blockingMethod ran during unblockedMethod
```

Reply:
fwiw, I do think there is something amiss because I've seen the warning quite a bit with `Meteor.callAsync`. Here's one repro: https://github.com/meteor/meteor/issues/12999. I haven't tested it with the latest `beta.6` though.

I've also seen it pop up during `Tinytest` runs.

Reply:
For me it appears for the simplest test you can do with a collection operation (a method behind the scenes):

Defined on client and server:
``` javascript
const NamedCollection = new Meteor.Collection('namedCollection');
NamedCollection.allow({
  insertAsync() { return true; },
  updateAsync() { return true; },
  removeAsync() { return true; },
});
```

And then run on the client:
``` javascript
await NamedCollection.insertAsync({ test: 1 });
```

Reply:
I can try to change finished=true as part of stubPromise resolution if you feel is correct, and provide the new tests that supposedly cover this scenario. I will do in another branch though, and the debug mode, even not used yet, can be merged as a new feature.

Reply:
> I can try to change finished=true as part of stubPromise resolution if you feel is correct

That is how it should work. It should also run the next queued function at the same time.

Reply:
Resolving this converstation. Debug mode will be added as a new future and covered in this PR.

For the other issue mentioned here, refer to this PR, https://github.com/meteor/meteor/pull/13055